mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/logistic_reg.rs

```diff
@@ -77,96 +215,122 @@ impl Worker for LogisticRegWorker {
-            logistic_reg_payload.input_model_columns,
-        )?;
-        self.input = Some(LogisticRegInput {
-            input_model_data: input,
-            target_model_data: target,
-            test_data,
+        let train_data_file_id = match file_ids.get(0) {
+            Some(value) => value.to_string(),
+            None => return Err(Error::from(ErrorKind::InvalidInputError)),
+        };
+
+        let target_data_file_id = match file_ids.get(1) {
+            Some(value) => value.to_string(),
+            None => return Err(Error::from(ErrorKind::InvalidInputError)),
+        };
+
+        self.input = Some(LogisticRegTrainWorkerInput {
+            train_alg_alpha: logisticreg_train_payload.train_alg_alpha,
+            train_alg_iters: logisticreg_train_payload.train_alg_iters,
+            train_data_file_id,
+            target_data_file_id,
         });
         Ok(())
     }
-    fn execute(&mut self, _context: WorkerContext) -> Result<String> {
+
+    fn execute(&mut self, context: WorkerContext) -> Result<String> {
         let input = self
             .input
             .take()
             .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
 
-        let mut logistic_mod = LogisticRegressor::default();
-        // Train the model
-        logistic_mod
-            .train(&input.input_model_data, &input.target_model_data)
+        let train_data_bytes = context.read_file(&input.train_data_file_id)?;
+        let target_data_bytes = context.read_file(&input.target_data_file_id)?;
+
+        let train_data_str = String::from_utf8(train_data_bytes)
             .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
-        // predict a new test data
-        let classes = logistic_mod
-            .predict(&input.test_data)
+        let target_data_str = String::from_utf8(target_data_bytes)
             .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
-        let mut output = String::new();
-        for c in classes.data().iter() {
-            writeln!(&mut output, "{}", c)
-                .map_err(|_| Error::from(ErrorKind::OutputGenerationError))?;
-        }
-        Ok(output)
+
+        let train_data_matrix = parse_input_to_matrix(&train_data_str.to_string())?;
+        let target_data_vector = parse_input_to_vector(&target_data_str.to_string())?;
+
+        let gd = GradientDesc::new(input.train_alg_alpha, input.train_alg_iters);
+        let mut logisticreg_train_mod = LogisticRegressor::new(gd);
+        //let mut logisticreg_train_mod = LogisticRegressor::default();
+        // train the model
+        logisticreg_train_mod
+            .train(&train_data_matrix, &target_data_vector)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+
+        // serialized the model
+        let model_json = serde_json::to_string(&logisticreg_train_mod).unwrap();
+
+        // save the model
+        context.save_file_for_all_participants(model_json.as_bytes())
```

Comment:
It's better to invoke "save_file_for_task_creator", because there is only one participant.